tests/cancun/eip6780_selfdestruct/test_create2_collision_selfdestruct.py

```diff
@@ -0,0 +1,129 @@
+    state_test: StateTestFiller,
+):
+    """Dynamic Create2->Suicide->Create2 collision scenario:"""
+    """
+    Perform a CREATE2, make sure that the initcode sets at least a couple of storage keys,
+    then on a different call, in the same tx, perform a self-destruct.
+    Then:
+        a) on the same tx, attempt to recreate the contract   <=== Covered int this test
+        b) on a different tx, attempt to recreate the contract
+    Perform a CREATE2, make sure that the initcode sets at least a couple of storage keys,
+    then in a different tx, perform a self-destruct.
+    Then:
+        a) on the same tx, attempt to recreate the contract
+        b) on a different tx, attempt to recreate the contract
+    Verify that the test case described
+    in https://wiki.hyperledger.org/pages/viewpage.action?pageId=117440824 is covered
+    """
+
+    address_to = to_address(0x0600)
+    address_s = to_address(0x0512)
+    suicide_d = to_address(0x03E8)
+
+    initcode = Initcode(
+        deploy_code=Op.SELFDESTRUCT(Op.PUSH20(suicide_d)),
+        initcode_prefix=Op.SSTORE(1, 1) + Op.CALL(Op.GAS(), Op.PUSH20(address_s), 0, 0, 0, 0, 0),
+    )
+    create2_address = compute_create2_address(address_to, 0, initcode)
+
+    pre = {
+        address_to: Account(
+            balance=1000000000000000000,
+            nonce=0,
+            code=Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+            + Op.MSTORE(0, Op.CREATE2(7, 0, Op.CALLDATASIZE(), 1))
+            + Op.CALL(Op.GAS(), Op.MLOAD(0), 3, 0, 0, 0, 0)
+            + Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+            + Op.SSTORE(1, Op.CREATE2(5, 0, Op.CALLDATASIZE(), 1)),
+            storage={0x01: 0xFF},
+        ),
+        address_s: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code=Op.SSTORE(1, 1),
+            storage={},
+        ),
+        TestAddress: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code="0x",
+            storage={},
+        ),
+    }
+
+    if create2_dest_already_in_state:
+        pre[create2_address] = Account(balance=13, nonce=1, storage={})
+
+    post = {
+        # Check what value has been transferred by suicide
+        suicide_d: Account(balance=10),
+        # Make sure the CREATE2 address performed storage operations
+        address_s: Account(storage={0x01: 0x01}),
+        # Make sure the second CREATE2 attempt is failing
+        address_to: Account(storage={0x01: 0x00}),
+    }
+
+    if create2_dest_already_in_state:
+        # Check that create2 account is unchanged
+        post[create2_address] = Account(balance=13, nonce=1, code="0x", storage={})
+    else:
+        # Make sure selfdestruct cleans the account
+        post[create2_address] = Account.NONEXISTENT
+
+    tx = Transaction(
+        ty=0x0,
+        chain_id=0x0,
+        nonce=0,
+        to=address_to,
+        gas_price=10,
+        protected=False,
+        data=initcode.bytecode,
```

Comment:
#### Error 
```
tests/cancun/eip6780_selfdestruct/test_create2_collision_selfdestruct.py:126: error: Argument "data" to "Transaction" has incompatible type "Optional[bytes]"; expected "Union[str, bytes, SupportsBytes, List[int]]"
```

#### Explanation
`Transaction.data` should be one of `str, bytes, SupportsBytes, List[int]` but `Initcode` inherits from `Code` which (unfortunately) allows `bytecode` to be of type `None`, which is not expected. 

#### Solution
This resolves the mypy issue, but is also burdensome and very ugly, as it implies in this test code that the bytecode could be `None`, which it never is.
```suggestion
        data=initcode.bytecode if initcode.bytecode is not None else bytes(),
```
 

#### Possible Improvement
I think we should make the default value for the optional `Transaction.data` compatible with the default value for the optional value for `Code.bytecode`.